apps/opik-backend/src/main/java/com/comet/opik/domain/ExperimentService.java

```diff
@@ -176,11 +176,20 @@ public Mono<Experiment> getById(@NonNull UUID id) {
 
                     return Mono.zip(
                             promptService.geyVersionsCommitByVersionsIds(promptVersionIds),
-                            Mono.fromCallable(() -> datasetService.findById(experiment.datasetId(), workspaceId))
+                            Mono.defer(() -> {
+                                try {
+                                    return Mono.just(Optional.of(datasetService.findById(
+                                            experiment.datasetId(), workspaceId)));
+                                } catch (NotFoundException e) {
+                                    return Mono.just(Optional.<Dataset>empty());
```

Comment:
What about creating another method returning Optional? So we don't need to have this exception

Reply:
I considered that option and couldn't decide which is best. If you prefer I can change this quite easily.

Reply:
OK done